src/client.ts

```diff
@@ -730,77 +727,62 @@ export class SpvWalletClient {
    * @example
    * // This function is a shorthand for:
    * const draft = await spvWalletClient.DraftToRecipients(recipients, metadata);
-   * const finalized = spvWalletClient.SignTransaction(draft);
+   * const finalized = await spvWalletClient.SignTransaction(draft);
    * const tx = await spvWalletClient.RecordTransaction(finalized, draft.id, metadata)
    */
   async SendToRecipients(recipients: Recipients, metadata: Metadata): Promise<Transaction> {
     const draft = await this.DraftToRecipients(recipients, metadata);
-    const finalized = this.SignTransaction(draft);
+    const finalized = await this.SignTransaction(draft);
     return this.RecordTransaction(finalized, draft.id, metadata);
   }
 
   /**
    * Finalize and sign the given draft transaction
    *
-   * @param {DraftTransaction} draftTransaction Draft transaction object
+   * @param {DraftTx} draftTransaction Draft transaction object
    * @return {string} Final transaction hex
-   */
-  SignTransaction(draftTransaction: DraftTransaction): string {
+   */  async SignTransaction(draftTransaction: DraftTx): Promise<string> {
     if (!this?.xPriv) {
       throw new ErrorNoXPrivToSignTransaction();
     }
 
-    const xPriv = this.xPriv as bsv.HDPrivateKey;
-    const txDraft: bsv.Transaction = new bsv.Transaction(draftTransaction.hex);
+    const xPriv = this.xPriv as HD;
```

Comment:
Isn't that type assertion redundant?
```suggestion
    const xPriv = this.xPriv
```